Demo/src/main/java/com/paypal/android/ui/paypal/PayPalNativeViewModel.kt

```diff
@@ -0,0 +1,207 @@
+        set(value) {
+            _uiState.update { it.copy(createdOrder = value) }
+        }
+
+    var completedOrder: Order?
+        get() = _uiState.value.completedOrder
+        set(value) {
+            _uiState.update { it.copy(completedOrder = value) }
+        }
+
+    var shippingPreference: ShippingPreferenceType
+        get() = _uiState.value.shippingPreference
+        set(value) {
+            _uiState.update { it.copy(shippingPreference = value) }
+        }
+
+    var payPalNativeCheckoutResult: PayPalNativeCheckoutResult?
+        get() = _uiState.value.payPalNativeCheckoutResult
+        set(value) {
+            _uiState.update { it.copy(payPalNativeCheckoutResult = value) }
+        }
+
+    var payPalNativeCheckoutError: PayPalSDKError?
+        get() = _uiState.value.payPalNativeCheckoutError
+        set(value) {
+            _uiState.update { it.copy(payPalNativeCheckoutError = value) }
+        }
+
+    private val payPalListener = object : PayPalNativeCheckoutListener {
+        override fun onPayPalCheckoutStart() {
+            isStartCheckoutLoading = true
+        }
+
+        override fun onPayPalCheckoutSuccess(result: PayPalNativeCheckoutResult) {
+            isStartCheckoutLoading = false
+            payPalNativeCheckoutResult = result
+        }
+
+        override fun onPayPalCheckoutFailure(error: PayPalSDKError) {
+            isStartCheckoutLoading = false
+            payPalNativeCheckoutError = error
+        }
+
+        override fun onPayPalCheckoutCanceled() {
+            isStartCheckoutLoading = false
+            isCheckoutCanceled = true
+        }
+    }
+
+    private val shippingListener = object : PayPalNativeShippingListener {
+
+        override fun onPayPalNativeShippingAddressChange(
+            actions: PayPalNativePaysheetActions,
+            shippingAddress: PayPalNativeShippingAddress
+        ) {
+            if (shippingAddress.adminArea1.isNullOrBlank() || shippingAddress.adminArea1 == "NV") {
+                actions.reject()
+            } else {
+                actions.approve()
+            }
+        }
+
+        override fun onPayPalNativeShippingMethodChange(
+            actions: PayPalNativePaysheetActions,
+            shippingMethod: PayPalNativeShippingMethod
+        ) {
+
+            viewModelScope.launch(exceptionHandler) {
+                orderId?.also {
+                    try {
+                        updateOrderUseCase(it, shippingMethod)
+                        actions.approve()
+                    } catch (e: IOException) {
+                        actions.reject()
+                        throw e
+                    }
+                }
+            }
+        }
+    }
+
+    private lateinit var payPalClient: PayPalNativeCheckoutClient
+    private lateinit var payPalDataCollector: PayPalDataCollector
+
+    private val exceptionHandler = CoroutineExceptionHandler { _, e ->
+        // TODO: propagate error to UI
```

Comment:
Yeah I wanted create an alert dialog here, but I haven't tried with Compose. There's a different API for composable dialogs that I wanted to hold off on until the entire merchant app is fully migrated to Compose. I can add a Toast here for now. In the long run this may turn into a broader discussion on how we expect merchants to handle exceptions in Kotlin.